tensorboard/plugins/core/core_plugin.py

```diff
@@ -102,20 +101,21 @@ def get_resource_apps(self):
 
     with self._assets_zip_provider() as fp:
       with zipfile.ZipFile(fp) as zip_:
-        # First collect the shasums that will later be used. Since hashes are
-        # in the hex, we convert it to base64 for CSP usage.
+
+        # First collect the shasums that will later be used.
         # TODO(stephanwlee): since index.html creates dynamic plugin and imports
         # their entry point (es_module_path), the CSP should include shasum
         # of those, too.
         shasums = {}
         for path in zip_.namelist():
-          if not path.startswith(SHASUM_DIR):
+          if os.path.splitext(path)[1] != '.html':
```

Comment:
I think we can combine the for loops now, right?  And avoid having a `shasums` dict at all.

```python
for path in zip_.namelist():
  # Do not serve the shasum data as static files.
  if path.startswith(SHASUM_DIR):
    continue
  gzipped_asset_bytes = _gzip(zip_.read(path))
  if os.path.splitext(path)[1] == '.html':
    checksum_path = os.path.join(SHASUM_DIR, path + SHASUM_FILE_SUFFIX)
    if checksum_path in zip_.namelist():
      shasums = zip_.read(checksum_path).splitlines(False)
    else:
      shasums = None
    wsgi_app = functools.partial(self._serve_html, shasums, gzipped_asset_bytes)
  else:
    wsgi_app = functools.partial(self._serve_asset, path, gzipped_asset_bytes)
  apps['/' + path] = wsgi_app

Reply:
done!